src/App.js

```diff
@@ -1,17 +1,197 @@
+        word:  words[rand].word,
+        hint: words[rand].hint,
+        secondHint : words[rand].secondHint
+      };
+    }
+    
+    secondHint = ()=>{
+      toShow = false;
+      alert(this.state.secondHint);
+      this.setState({...this.state});
+    }
+    
+    removeLetter = () => {
+      let flag = 1;
+      toShow = false;
+      let newState =  {...this.state};
+      let letterStatus = newState.letterStatus;
+      Object.keys(letterStatus).map(l=>{
+        if(flag&&(!letterStatus[l])&&(!this.state.word.includes(l))){
+          letterStatus[l]=true;
+          flag = 0;
+          console.log(l,this.state.word.includes(l));
+        }
+    });
+    this.setState({...this.state,
+      letterStatus:newState.letterStatus,
+    });
   }
-}
 
-export default App;
+     
+    resetGame = ()=> {
+      toShow = false;
+      let r = Math.floor((Math.random() * words.length));
+      let word = words[r].word;
+      let hint = words[r].hint;
+      if(words.length>1){
+        words.splice(r,1);
+      }
+      let newState = {...this.state};
+      newState.word = word;
+      newState.hint = hint;
+      newState.score ={value: 100, className: "high-score"};
+      newState.letterStatus = this.generateLetterStatus();
+      this.setState({...this.state,word:newState.word,
+        score:newState.score,
+        letterStatus:newState.letterStatus,
+        hint:newState.hint
+      });
+    }
+    
+    resetLife = ()=> {
+      toShow = false; //hide
+      let newState = {...this.state};
+      newState.score ={value: 100, className: "high-score"};
+      newState.letterStatus = this.generateLetterStatus();
+      this.setState({...this.state,
+        score:newState.score,
+        letterStatus:newState.letterStatus
+      });
+    }
+    
+    generateLetterStatus() {
+      let letterStatus = {};
+      for (let i = 65; i < 91; i++) {
+        letterStatus[String.fromCharCode(i)] = false;
+      }
+      return letterStatus;
+    }
+    
+    selectLetter = letter => {
+      let state =  { ...this.state}
+      let letterStatus = state.letterStatus;
+      //let letterStatus = this.state.letterStatus - > never use it like this
+      letterStatus[letter] = true;
+      //state.word.indexOf(letter) > -1 ? this.updateScore(5): this.updateScore(-20);
+      (this.state.word.includes(letter))?this.updateScore(5):this.updateScore(-20);
+      this.setState({ letterStatus: letterStatus }); //the right-hand side is the updated object! 
+    };
+    
+    //return new object - just one update the status
+    //For each right letter, increase the score by 5
+    //For each wrong letter, decrease the score by 20
+    updateScore = update => {
+      let newScore = { ...this.state.score };
+      newScore.value += update;
+      if (newScore.value >= 80) {
+        toShow = false; //hide
+        newScore.className = "high-score";
+        this.setState({ ...this.state, score: newScore });
```

Comment:
move this line to the end of the `if-else` block, remove it from the other blocks